### PR TITLE
Enhancement 2506: Add new profiling API

### PR DESCRIPTION
**Pull request by hhellyer**
## What type of PR is this ? 

- [ ] Bug fix
- [X] Enhancement

## What does this PR do ?

 - Adds profiling rest APIs to allow the UI to run JSONPath queries against our profiling data.
 - Adds post processing scripts for node profiling data.
 - Invokes post processing after profiling is complete.
 - Adds tests and test data.

The REST API's use POST rather than GET as GET requests are not expected to have a body. Proxy's between PFE and the user may drop the body from a GET request. (And the bug would be ours to fix.) An alternative would be to pass the JSONPath query in the query field of the URL but that may be ugly as it will contain many special characters.

## Which issue(s) does this PR fix ?
This is the first part of implementing:
https://github.com/eclipse/codewind/issues/2506

The next part will be adding the ability to post process Java profiling data (in .hcd) files to the standard format we will be using so the UI can work with 

#### Link to the [Codewind repository](https://github.com/eclipse/codewind/issues) issue(s) this PR fixes or references:
https://github.com/eclipse/codewind/issues/2506

## Does this PR require a documentation change ?
No - the openapi.yml for our REST API changes but we do not expect users to interact directly with this feature.

### CODE REVIEW

File: src/pfe/portal/modules/LoadRunner.js [hunk 1]
```diff
@@ -636,6 +645,26 @@ module.exports = class LoadRunner {
...
+  async postProcessProfilingData() {
+    /* Run the merge out of process as the profiling data could potentially be quite large. */
+    const profilingPath = path.join(this.workingDir + '/profiling.json');
+    const normalisedProfilingPath = this.project.getPathToProfilingTreeFile(this.metricsFolder);
+    const mergeCommand = `${process.execPath} /portal/scripts/mergeNodeProfiles.js ${profilingPath} ${normalisedProfilingPath}`;
+    log.debug(`Running: ${mergeCommand}`);
+    const mergeProcess = await exec(mergeCommand);
+    log.trace(mergeProcess.stdout);
+    log.trace(mergeProcess.stderr);
+    log.debug('Merging done');
```
Comment (markcor11) [inline src/pfe/portal/modules/LoadRunner.js:657]: Nice debug

File: src/pfe/portal/modules/LoadRunner.js [hunk 2]
```diff
@@ -636,6 +645,43 @@ module.exports = class LoadRunner {
...
+
+    const summarisedProfilingPath = this.project.getPathToProfilingSummaryFile(this.metricsFolder);
+    const summariseCommand = `${process.execPath} /portal/scripts/summariseProfile.js ${normalisedProfilingPath} ${summarisedProfilingPath}`;
+    log.debug(`Running: ${summariseCommand}`);
+    try {
+      const summariseProcess = await exec(mergeCommand);
+      // Log these but they should be empty in the success case.
+      log.trace(summariseProcess.stdout);
+      log.trace(summariseProcess.stderr);
+      log.debug('Merging done');
```
Comment (tobespc) [inline src/pfe/portal/modules/LoadRunner.js:675]: if there is an error in merging, should we still say 'merging done' ?

File: src/pfe/portal/modules/LoadRunner.js (hunk 1 above)
Comment (hhellyer) [reply src/pfe/portal/modules/LoadRunner.js]: That should result in an exception and triggering of the `catch` block.
I've realised the message from the summarising block below says `merging` so I'll push a fix for that.

Comment (hhellyer) [reply src/pfe/portal/modules/LoadRunner.js]: That should result in an exception and triggering of the `catch` block.
I've realised the message from the summarising block below says `merging` so I'll push a fix for that.

**Pull request merged**